apps/smtp/src/modules/smtp/email-compiler.ts

```diff
@@ -0,0 +1,132 @@
+import { compileMjml } from "./compile-mjml";
+import { compileHandlebarsTemplate } from "./compile-handlebars-template";
+import { MessageEventTypes } from "../event-handlers/message-event-types";
+import { htmlToPlaintext } from "./html-to-plaintext";
+import { SmtpConfiguration } from "./configuration/smtp-config-schema";
+import { createLogger } from "../../logger";
+
+interface CompileArgs {
+  smtpConfiguration: SmtpConfiguration;
+  recipientEmail: string;
+  event: MessageEventTypes;
+  payload: any;
+}
+
+export interface CompiledEmail {
+  from: string;
+  to: string;
+  text: string;
+  html: string;
+  subject: string;
+}
+
+export interface IEmailCompiler {
+  compile(args: CompileArgs): CompiledEmail | undefined;
+}
+
+/*
+ * todo introduce modern-errors
+ */
+export class EmailCompiler implements IEmailCompiler {
+  constructor() {}
+
+  compile({
+    payload,
+    recipientEmail,
+    event,
+    smtpConfiguration,
+  }: CompileArgs): CompiledEmail | undefined {
+    const logger = createLogger("sendSmtp", {
+      name: "sendSmtp",
+      event,
+    });
+
+    const eventSettings = smtpConfiguration.events.find((e) => e.eventType === event);
+
+    if (!eventSettings) {
+      logger.debug("No active settings for this event, skipping");
```

Comment:
suggestion: if customers have broken config we won't know about this if we use `debug` level

```suggestion
      logger.info("No active settings for this event, skipping");
```